# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: docshell/base/WindowContext.cpp
# Commit: 80fed0cf581c
# Full Hash: 80fed0cf581ca575f6dc90714834650f156bf1e5
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2020-04-30 03:51:35
# Description:
#   Bug 1633204 - Live WindowContexts should keep BrowsingContextGroup alive, r=farre
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D72893
# ==============================================================================

diff -r 8c1e8723a695 -r 80fed0cf581c docshell/base/WindowContext.cpp
--- a/docshell/base/WindowContext.cpp	Wed Apr 29 18:37:37 2020 +0000
+++ b/docshell/base/WindowContext.cpp	Wed Apr 29 18:49:10 2020 +0000
@@ -154,6 +154,7 @@
 
   // Register this to the browsing context.
   mBrowsingContext->RegisterWindowContext(this);
+  Group()->Register(this);
 }
 
 void WindowContext::Discard() {
@@ -167,6 +168,7 @@
   mIsDiscarded = true;
   gWindowContexts->Remove(InnerWindowId());
   mBrowsingContext->UnregisterWindowContext(this);
+  Group()->Unregister(this);
 }
 
 WindowContext::WindowContext(BrowsingContext* aBrowsingContext,